### DISCUSSION TITLE
"Ping Pong" GlobalScope Function

### DISCUSSION THREAD

**Discussion by FireCatMagic** [godotengine/godot-proposals#9857] [Engine Core]
Like "wrap" and "wrapf", but "ping pongs" the value. I'm not sure of a better name for it, but there probably is one.
Example

ping_pong(value, min, max)

This would return a value that approaches max, and then when it gets to max, it starts going back towards min. Basically goes back and forth between the two values.

example:
ping_pong(1.6, 0.0, 1.0)
Returns 0.4
It reaches 1.0, then starts going backwards 0.6, landing at 0.4.

If this function was wrapf, it would return 0.6, because it wraps back to zero and goes from there. But for the ping pong function, it alternates between starting at the maximum and minimum.

**Comment by AThousandShips** [comment]
You mean like [`pingpong`](https://docs.godotengine.org/en/latest/classes/class_%40globalscope.html#class-globalscope-method-pingpong)? You just need to do `min + ping-pong(value - min, max - min)` for your case

**Reply by FireCatMagic** [reply]
oh i'm stupid it already exists 💀

**Reply by AThousandShips** [reply]
Is there anything else you're wondering or can this be closed? The feature exists

**Reply by FireCatMagic** [reply]
you can close it
